pallets/msa/src/lib.rs

```diff
@@ -175,6 +196,29 @@ pub mod pallet {
 	pub(super) type MsaInfoOf<T: Config> =
 		StorageMap<_, Twox64Concat, MessageSourceId, u8, ValueQuery>;
 
+	/// Storage type for mapping a "bucket" number to a mortality block
+	/// Used to know when to clear a virtual "bucket" of stored signatures
+	/// in PayloadSignatureRegistry
+	#[pallet::storage]
+	#[pallet::getter(fn get_mortality_block_of)]
+	pub type MortalityBlockOf<T: Config> =
+		StorageMap<_, Twox64Concat, T::BlockNumber, T::BlockNumber, OptionQuery>;
```

Comment:
I'm obviously lacking some context but it seems like if the `MortalityBucketSize` is fixed then you can always calculate MortalityBlockOf certain block. Unless something dynamically modifies it

Reply:
That is basically a registry for each bucket's current mortality block.  It gets changed every MortalityBucketSize * NumberOfBuckets (or longer if the extrinsics that check it are not called).